### DISCUSSION TITLE
Will client.authStore.onChange trigger from side effects such as backend changes?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#861] [Q&A] [answered]
Just wondering if `onChange` is meant to fire only from frontend actions?

For example, if I change a user to`verified=false` in the database, will that fire an `onChange` event on the frontend? Are there other similar cases where side effects might trigger it, or is it always just following frontend actions?

**Comment by ganigeorgiev** [comment, marked as answer]
No. The `AuthStore` is changed only after some frontend action. In v0.8 the users will be now a regular collection so you'll be able to subscribe to realtime changes of the authenticated model, but this initially will not be binded by default, aka. if users want this behavior they will have do something like:

```js
client.collection("users").subscribeOne(client.authStore.model.id, (e) => {
    if (client.authStore.model?.id === e.record.id) {
        if (e.action === 'update') {
            client.authStore.save(client.authStore.token, e.record);
        } else if (e.action === 'delete') {
            client.authStore.clear();
            client.collection("users").unsubscribe(e.record.id);
        }
    }
});
```

**Reply by benallfree** [reply]
I don't understand this example because those methods don't exist on `client`. Is this the new v0.8 client?

**Reply by ganigeorgiev** [reply]
Yes, sorry I forgot to mention it. Most likely tomorrow I'll push the pre-release and will be publicly available.

**Comment by benallfree** [comment]
Thanks. I noticed that the model is stored in `localStorage` I have to use `client.users.refresh()` to manually reverify with the backend when my frontend boots, then watch for realtime changes.

Are there plans to create a more realtime connection? It's a little unexpected that `BaseAuthStore.onChange` doesn't fire when backend changes are made to the user model.

**Reply by ganigeorgiev** [reply]
> Are there plans to create a more realtime connection?

As mentioned in the previous comment, you'll be able to listen to realtime changes of the auth model(s), but it will be up to the users to decide whether they want to initialize it or not. Depending on the users demand for this, for simplicity we may add in the future a helper similar to the above example, eg. `client.realtimeAuthSync(bool)` or similar.
